ariadne/cardano/src/Ariadne/Wallet/Backend/Util.hs

```diff
@@ -0,0 +1,103 @@
+module Ariadne.Wallet.Backend.Util
+       ( allAccountIds
+       , ourAccountIdFromAccounts
+       , accountsToUse
+       ) where
+
+import Control.Lens (ix)
+import qualified Data.Text.Buildable
+import Formatting (bprint, build, int, (%))
+
+import Pos.Util (maybeThrow)
+
+import Ariadne.Wallet.Backend.KeyStorage (resolveWalletRefThenRead)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet
+  (HdAccount, HdAccountId, HdRootId, hdAccountId, hdAccountIdParent)
+import Ariadne.Wallet.Cardano.Kernel.DB.HdWallet.Read (readAccountsByRootId)
+import Ariadne.Wallet.Cardano.Kernel.DB.Util.IxSet (IxSet, (@+))
+import Ariadne.Wallet.Cardano.WalletLayer (PassiveWalletLayer(..))
+import Ariadne.Wallet.Face
+  (LocalAccountReference(..), WalletReference, WalletSelection(WSAccount))
+
+
+data OurAccountException = OurAccountNoAccounts !HdRootId
+                         | OurAccountIndexOutOfRange !HdRootId !Word
+                       deriving (Show)
+
+instance Buildable OurAccountException where
+    build = \case
+        OurAccountNoAccounts walletIdx ->
+            bprint ("Wallet "%build%" does not have any accounts") walletIdx
+        OurAccountIndexOutOfRange rootId idx ->
+            bprint ("Account #"%int%" doesn't exist in "%build) idx rootId
+
+instance Exception OurAccountException where
+    displayException = toString . prettyL
+
+-- | Used for fee estimate
+allAccountIds :: PassiveWalletLayer IO
+             -> IORef (Maybe WalletSelection)
+             -> WalletReference
+             -> [LocalAccountReference]
+             -> IO [HdAccountId]
+allAccountIds pwl walletSelRef walletRef accRefs =
+         fmap (^. hdAccountId) . toList . snd
+     <$> accountsToUse pwl walletSelRef walletRef accRefs
+
+-- | Used for sending tx with existing accountsToUse
+ourAccountIdFromAccounts :: MonadThrow m
```

Comment:
1. I would rename it to `changeAccountIdFromAccount` and `ourAccount{,Id}` to `changeAccount{,Id}`.
2. After recent changes this function is only used in `Tx.hs`, I propose to keep it there.